kernel/src/vm/vmar/mod.rs

```diff
@@ -1384,4 +1385,143 @@ mod test {
             (_, None)
         ));
     }
+
+    /// Helper function to create a dummy `IoMem` for testing.
+    #[expect(unsafe_code)]
+    fn create_dummy_iomem(paddr: usize, size: usize) -> IoMem {
```

Comment:
Remove this. This isn't safe at all. Use `IoMem::acquire` instead. You may need to choose a larger address to ensure that it succeeds (it should be larger than the maximum physical address).